client/java/generator/src/main/java/io/openlineage/client/TypeResolver.java

```diff
@@ -137,7 +138,15 @@ public ResolvedType visit(ArrayType arrayType) {
 
         @Override
         public ResolvedType visit(OneOfType oneOfType) {
-          throw new UnsupportedOperationException("oneOf is not supported yet " + oneOfType);
+          List<ResolvedType> resolvedTypes = oneOfType
+              .getTypes()
+              .stream()
+              .filter(type -> type instanceof RefType)
```

Comment:
That's kind of workaround for: 
```
"oneOf": [
    { "$ref": "#/$defs/RunEvent" },
    { "$ref": "#/$defs/DatasetEvent" },
    { "$ref": "#/$defs/JobEvent" }
  ]
```

We don't want to generate single Java class to handle that (may be undoable), but we want to generate code for each of types which is `RefType`. The method returns the first of the resolved types because it has to align with 
```
public ResolvedType visit(...) {
```
contract and changing contract to a list or optional would require changes across a whole code generator.